tensorboard/backend/application.py

```diff
@@ -265,7 +266,32 @@ def __init__(self, plugins, path_prefix=''):
         else:
           path = (self._path_prefix + DATA_PREFIX + PLUGIN_PREFIX + '/' +
                   plugin.plugin_name + route)
-        self.data_applications[path] = app
+
+        if path.endswith('/*'):
+          # Note we remove the '*' but leave the slash in place.
+          path = path[:-1]
+          if '*' in path:
+            # note we re-add the removed * in the format string
+            raise ValueError('Plugin %r handles invalid route %r*:  Only '
+                             'trailing wildcards are supported '
+                             '(i.e., `/.../*`)' %
+                             (plugin.plugin_name, path))
+          unordered_prefix_routes[path] = app
+        else:
+          if '*' in path:
+            raise ValueError('Plugin %r handles invalid route %r:  Only '
+                             'trailing wildcards are supported '
+                             '(i.e., `/.../*`)' %
+                             (plugin.plugin_name, path))
+          self.exact_routes[path] = app
+
+    # Wildcard routes will be checked in the given order, so we sort them
+    # longest to shortest so that a more specific route will take precedence
+    # over a more general one (e.g., a catchall route `/*` should come last).
+    self.prefix_routes = collections.OrderedDict(
+        sorted(six.iteritems(unordered_prefix_routes),
```

Comment:
nit: [Google Python style says][1] that when “using a hanging indent of
4 spaces […] there should be nothing after the open parenthesis or
bracket on the first line”.

[1]: https://google.github.io/styleguide/pyguide.html#34-indentation

```python
    self.prefix_routes = collections.OrderedDict(
        sorted(
            six.iteritems(unordered_prefix_routes),
            key=lambda x: len(x[0]),
            reverse=True))
```

Reply:
done